addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -256,7 +257,12 @@ class MessageList extends Component {
                 )
             )
             .map(([refId, ref]) => ref)
-            .sort((ref1, ref2) => (ref1.message.id < ref2.message.id ? -1 : 1));
+            .sort((ref1, ref2) => {
+                if (ref1.message.id < 0 || ref2.message.id < 0) {
+                    return (ref1.message.id < ref2.message.id ? 1 : -1);
+                }
+                return (ref1.message.id < ref2.message.id ? -1 : 1);
```

Comment:
There is too much magic going on here.
Code would be much more readable if all 4 cases (of id < 0 and id > 0) where explicit.